node/src/accountant/payable_dao.rs

```diff
@@ -559,66 +725,130 @@ mod tests {
+        let rowid_2 = 792;
+        let previous_timestamp_2 = from_time_t(187_100_000);
+        let new_payable_timestamp_2 = from_time_t(191_333_000);
+        let starting_amount_2 = 202_000_000;
+        let payment_2 = 200_300_400;
+        let wallet_2 = make_wallet("booble bobble");
+        {
+            create_payable_account_with_pending_payment(
+                conn,
+                &wallet_1,
+                starting_amount_1,
+                previous_timestamp_1,
+                rowid_1,
+            );
+            create_payable_account_with_pending_payment(
+                conn,
+                &wallet_2,
+                starting_amount_2,
+                previous_timestamp_2,
+                rowid_2,
+            )
+        }
+        let fingerprint_1 = PendingPayableFingerprint {
+            rowid: rowid_1,
+            timestamp: new_payable_timestamp_1,
+            hash: hash_1,
+            attempt: 1,
+            amount: payment_1,
+            process_error: None,
+        };
+        let fingerprint_2 = PendingPayableFingerprint {
+            rowid: rowid_2,
+            timestamp: new_payable_timestamp_2,
+            hash: hash_2,
+            attempt: 1,
+            amount: payment_2,
+            process_error: None,
+        };
+        TestSetupValuesHolder {
+            fingerprint_1,
+            fingerprint_2,
+            wallet_1,
+            wallet_2,
+            starting_amount_1,
+            starting_amount_2,
+        }
+    }
+
     #[test]
     fn transaction_confirmed_works() {
         let home_dir =
             ensure_node_home_directory_exists("payable_dao", "transaction_confirmed_works");
         let boxed_conn = DbInitializerReal::default()
             .initialize(&home_dir, DbInitializationConfig::test_default())
             .unwrap();
-        let hash = H256::from_uint(&U256::from(12345));
-        let rowid = 789;
-        let previous_timestamp = from_time_t(190_000_000);
-        let payable_timestamp = from_time_t(199_000_000);
-        let attempt = 5;
-        let starting_amount = 10000;
-        let payment = 6666;
-        let wallet = make_wallet("bobble");
-        {
-            insert_record_fn(
-                &*boxed_conn,
-                &wallet.to_string(),
-                starting_amount,
-                to_time_t(previous_timestamp),
-                Some(sign_conversion::<u64, i64>(rowid).unwrap()),
-            );
-        }
+        let setup_holder =
+            make_fingerprint_pair_and_insert_initial_payable_records(boxed_conn.as_ref());
         let subject = PayableDaoReal::new(boxed_conn);
-        let pending_payable_fingerprint = PendingPayableFingerprint {
-            rowid_opt: Some(rowid),
-            timestamp: payable_timestamp,
-            hash,
-            attempt_opt: Some(attempt),
-            amount: payment,
-            process_error: None,
+        let expected_account_1 = PayableAccount {
```

Comment:
Not a big thing, but since this is really part of the Assert section of the test, it would be nice to get it out of the Arrange section. If ownership problems prevent you, don't worry about it.